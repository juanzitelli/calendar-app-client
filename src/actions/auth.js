import Swal from "sweetalert2";
import { fetchConToken, fetchSinToken } from "../helpers/fetch";
import { types } from "../types/types";

export const startLogin = (email, password) => async dispatch => {
  const respuesta = await fetchSinToken("auth", { email, password }, "POST");
  const body = await respuesta.json();
  if (body.ok) {
    localStorage.setItem("x-token", body.token);
    localStorage.setItem("token-init-date", new Date().getTime());
    dispatch(login({ uid: body.uid, name: body.name }));
  } else {
    Swal.fire("Error", body.msg, "error");
  }
};

export const login = user => ({
  type: types.authLogin,
  payload: user,
});

export const startRegister = (email, password, name) => async dispatch => {
  const respuesta = await fetchSinToken(
    "auth/new",
    { email, password, name },
    "POST"
  );
  const body = await respuesta.json();
  if (body.ok) {
    localStorage.setItem("x-token", body.token);
    localStorage.setItem("token-init-date", new Date().getTime());
    dispatch(login({ uid: body.uid, name: body.name }));
  } else {
    Swal.fire("Error", body.msg, "error");
  }
};

export const startChecking = () => async dispatch => {
  const respuesta = await fetchConToken("auth/renew");
  const body = await respuesta.json();
  if (body.ok) {
    localStorage.setItem("x-token", body.token);
    localStorage.setItem("token-init-date", new Date().getTime());
    dispatch(login({ uid: body.uid, name: body.name }));
  } else {
    dispatch(checkingFinished());
  }
};

const checkingFinished = () => ({
  type: types.authCheckingFinished,
});

export const startLogout = () => async dispatch => {
  localStorage.clear();
  dispatch(logout());
};

export const logout = () => ({
  type: types.authLogout,
});
