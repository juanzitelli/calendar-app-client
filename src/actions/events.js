import Swal from "sweetalert2";
import { fetchConToken } from "../helpers/fetch";
import { prepareEvents } from "../helpers/prepareEvents";
import { types } from "../types/types";

export const eventAddNew = event => ({
  type: types.eventAddNew,
  payload: event,
});

export const startEventAddNew = event => async (dispatch, getState) => {
  const { uid, name } = getState().auth;
  try {
    const resp = await fetchConToken("events", event, "POST");
    const body = await resp.json();

    if (body.ok) {
      event.id = body.evento.id;
      event.user = {
        _id: uid,
        name,
      };
      dispatch(eventAddNew(event));
    }
  } catch (error) {
    console.error(error);
  }
};

export const eventSetActive = event => ({
  type: types.eventSetActive,
  payload: event,
});

export const eventClearActiveEvent = () => ({
  type: types.eventClearActiveEvent,
});

export const startEventUpdate = event => async dispatch => {
  try {
    const resp = await fetchConToken(`events/${event.id}`, event, "PUT");
    const body = await resp.json();
    if (body.ok) {
      dispatch(eventUpdated(event));
    } else {
      Swal.fire("Error", body.msg, "error");
    }
  } catch (error) {
    console.error(error);
  }
};

export const eventUpdated = event => ({
  type: types.eventUpdated,
  payload: event,
});

export const startEventDelete = () => async (dispatch, getState) => {
  const { id } = getState().calendar.activeEvent;
  try {
    const resp = await fetchConToken(`events/${id}`, {}, "DELETE");
    const body = await resp.json();
    if (body.ok) {
      dispatch(eventDeleted());
    } else {
      Swal.fire("Error", body.msg, "error");
    }
  } catch (error) {
    console.error(error);
  }
};

const eventDeleted = () => ({
  type: types.eventDeleted,
});

export const eventsStartLoading = () => async dispatch => {
  try {
    const response = await fetchConToken("events");
    const { eventos } = await response.json();
    
    dispatch(eventsLoaded(prepareEvents(eventos)));
  } catch (error) {
    console.error(error);
  }
};

const eventsLoaded = events => ({
  type: types.eventsLoaded,
  payload: events,
});

export const eventsLogout = () => ({
  type: types.eventLogout,
})
