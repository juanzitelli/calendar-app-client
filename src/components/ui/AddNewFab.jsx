import React from 'react'
import { useDispatch } from 'react-redux'
import { uiOpenModalAction } from '../../actions/ui'

export const AddNewFab = () => {
	const dispatch = useDispatch()
	const onClickEventHandler = () => {
		dispatch(uiOpenModalAction());
	}
	return (
		<button className="btn btn-primary fab" onClick={onClickEventHandler}>
			<i className="fas fa-plus"></i>
		</button>
	)
}
