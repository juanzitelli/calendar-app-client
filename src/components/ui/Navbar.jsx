import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startLogout } from '../../actions/auth';
import { eventsLogout } from '../../actions/events';

export const Navbar = () => {
	const { name } = useSelector(state => state.auth)
	const dispatch = useDispatch();
	const handleLogout = () => {
		dispatch(startLogout())
		dispatch(eventsLogout());
	}
	return (
		<div className="navbar navbar-dark bg-dark">
			<span className="navbar-brand">
				{name}'s calendar ğŸ“…
			</span>
			<button className="btn btn-outline-danger">

				<span onClick={handleLogout}><i className="fas fa-sign-out-alt"></i> Exit</span>
			</button>
		</div>
	)
}
