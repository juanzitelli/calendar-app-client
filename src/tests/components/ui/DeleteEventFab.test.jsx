import { mount } from 'enzyme';
import React from 'react'
import { Provider } from 'react-redux';
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { startEventDelete } from '../../../actions/events';
// import { startEventDelete } from '../../../actions/events';



const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {};
let store = mockStore(initState);
jest.mock("../../../actions/events/", () => ({
	startEventDelete: jest.fn(),
}))

store.dispatch = jest.fn();


describe('Tests on DeleteEventFab component', () => {
	const wrapper = mount(<Provider store={store} >
		<DeleteEventFab />
	</Provider>)
	beforeEach(() => {
		store = mockStore(initState);
		jest.clearAllMocks();
	});
	test('Should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})
	test('Should call eventStartDelete on click event', () => {
		wrapper.find("button").prop('onClick')();
		expect(startEventDelete).toHaveBeenCalled();
	})

});


