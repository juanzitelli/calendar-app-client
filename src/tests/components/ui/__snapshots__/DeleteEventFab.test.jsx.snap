// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on DeleteEventFab component Should match snapshot 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Delete event
  </span>
</button>
`;
